.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,42 @@
+name: Integration Tests
+
+on:
+  pull_request:
+    branches:
+      - '**'  # Matches all branch names, for PRs
+  push:
+    branches:
+      - 'main'  # Run tests on main branch
+
+jobs:
+  integration-tests:
+    name: integration-tests
+    runs-on: ${{matrix.os}}
+    timeout-minutes: 5
```

Comment:
bruh I was just about to ask you to add this, but looks like you're already ahead of me! 😄